frontend/app/services/letters-service.server.ts

```diff
@@ -0,0 +1,46 @@
+import { z } from 'zod';
+
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const letterSchema = z.object({
+  dateSent: z.date().optional(),
+  letterTypeCode: z.string().optional(),
+  referenceId: z.string().uuid().optional(),
+});
+
+export type LettersInfo = z.infer<typeof letterSchema>;
+
+function createLettersService() {
+  const logger = getLogger('letters-service.server');
+  const { INTEROP_API_BASE_URI } = getEnv();
+
+  /**
+   *
+   * @param { userId }
+   * @returns returns the letters based off  @param userId
+   */
+  async function getLetters(requestBody: { userId: string }) {
+    const url = `${INTEROP_API_BASE_URI}/letters?userId=${requestBody.userId}`;
+    const response = await fetch(url);
+
+    if (response.ok) return letterSchema.parse(await response.json());
+    if (response.status === 404) return null;
+
+    logger.error('%j', {
+      message: 'Failed to fetch data',
+      status: response.status,
+      statusText: response.statusText,
+      url: url,
+      responseBody: await response.text(),
+    });
+
+    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
```

Comment:
I would handle the error state earlier in the function. If there's a problem with the response, return early.

This then gives you the freedom afterwards to post-process the response data however you see fit. Granted, in this case, there's no post-processing happening.. but who knows what future iterations of this code might look like.

Reply:
Got it. Fixed: c20e06602530ac9a3fb8cf2b080a3b3fc36d0451.